src/components/Teaser/Teaser.tsx

```diff
@@ -1,104 +1,182 @@
+        <Text inline title={text}>
+            {text}
+        </Text>
+    );
+
+    if (icon) {
+        return (
+            <div {...elem(['row', 'sourceContainer'])}>
+                {React.cloneElement(icon, { ...elem('sourceIcon') })}
+                <Text inline title={text} {...elem('caption')}>
+                    {text}
+                </Text>
+            </div>
+        );
+    }
+
+    if (href) {
+        return (
+            <Link href={href} {...elem('link')}>
+                {textComponent}
+            </Link>
+        );
+    }
+
+    return React.cloneElement(textComponent, { ...elem('info') });
+};
+
 export const Teaser: React.FC<Props> = ({
     title,
-    subTitle = '',
-    location = '',
-    details = '',
-    statuses,
-    disabled = false,
+    subtitle,
+    hasCheckbox = false,
+    id,
+    isSelected,
+    onChange,
+    primaryInfo,
+    secondaryInfo,
+    tercearyInfo,
+    sourceInfo,
+    isVisited,
+    timestamp,
+    tags,
+    matchingIndicatorPercentage,
     ...rest
 }) => {
-    const subTitleElement = (
-        <Text
-            inline
-            title={typeof subTitle === 'string' ? subTitle : undefined}
-            context={disabled ? 'neutral' : 'background'}
-            {...elem('subTitle')}
-        >
-            {subTitle}
-        </Text>
-    );
-    const detailsElement = (
-        <Text
-            inline
-            title={typeof details === 'string' ? details : undefined}
-            context="neutral"
-            {...elem('details')}
-        >
-            {details}
-        </Text>
-    );
+    const useId = React.useId();
+    const checkboxId = id ?? useId;
+
+    const computeMatchingIndicatorPercentageColor = (value: number) => {
+        if (value <= 30) {
+            return 'var(--color-icon-critical-default)';
+        }
+        if (value <= 50) {
+            return 'var(--color-icon-caution-default)';
+        }
+        return 'var(--color-icon-success-default)';
+    };
+
     return (
         <div {...rest} {...block({ ...rest })}>
-            <div {...elem('line')}>
-                <Text
-                    inline
-                    context={disabled ? 'neutral' : 'foreground'}
-                    {...elem('title')}
-                    title={typeof title === 'string' ? title : undefined}
-                >
-                    {title}
-                </Text>
-                {location && (
-                    <Text inline context="neutral" {...elem('location')}>
-                        {location}
+            {hasCheckbox && <Checkbox id={checkboxId} checked={isSelected} onChange={onChange} />}
+            <div {...elem('column')}>
+                <div {...elem(['row', 'titleRow'])}>
+                    <div {...elem('row')}>
+                        <Text inline title={title} {...elem('title')}>
+                            {title}
+                        </Text>
+                        {matchingIndicatorPercentage !== undefined &&
```

Comment:
maybe this could be a separate component? I can see how we might want to reuse it at some point...